pageserver/src/page_service.rs

```diff
@@ -942,6 +950,13 @@ where
     ) -> Result<(), QueryError> {
         let ctx = self.connection_ctx.attached_child();
         debug!("process query {query_string:?}");
+        fail::fail_point!(
+            "simulated-bad-compute-connection",
+            rand::random::<f64>() < CONNECTION_FAILURE_PROBABILITY,
+            |_| Err(QueryError::Disconnected(ConnectionError::Io(
+                io::Error::new(io::ErrorKind::ConnectionReset, "Simulated connection error")
+            )))
```

Comment:
Regardless of my recent https://github.com/neondatabase/neon/pull/5696 I think we are going to be flushing for these connections. Ugh, I so wish we had removed this anonymous inner class alike PostgresQueryHandler already because it is making everything more difficult to see. Also, solving that good will be a difficult one.

Perhaps we could in addition to returning an error also cancel the cancellation token? That would give us fast exit. If we are always using `task_mgr::shutdown_token` for shutdown, then it will be fine to cancel. We most likely are using it.

Reply:
Looks like cancellation token isn't making it actually disconnect.

Reply:
Even if you cancel the token within the failpoint? It is very difficult to eyeball this. Perhaps we do need to actually kill the err the connection, but at the same time I would not expect you to be able to return a connection reset error here if nothing useful was done to it.

(I wrote #5733 in the meantime about this big picture problem with this abstraction.) 

We could have a failpointable TcpStream wrapper which is later composed to `tokio_io_timeout`, which would drop the actual connection as perhaps the layers of abstraction is just too much on this side.

Reply:
OK i just decided to have a failpoint-configurable average timeout that follows an exponential distribution to test it. This does seem to create a lot of connection errors on the compute side, and does degrade performance (as expected. It usually executes 1500 times with a good connection and now executes 600 times with a bad one), but the workload does still seem to execute successfully